File: server/controllers/gpt.js
const { Configuration, OpenAIApi } = require("openai");
require("dotenv").config();

const configuration = new Configuration({
  apiKey: process.env.OPENAI_API_KEY,
});
const openai = new OpenAIApi(configuration);

const gptGetResponse = async (cmd) => {
  const completion = await openai.createChatCompletion({
    model: "gpt-3.5-turbo",
    messages: [{ role: "assistant", content: cmd }],
  });
  console.log(completion);
  // return completion.data.choices[0].message;
  return 'hello';
};

const gptGetTitle = async (topic) => {
  try {
    const completion = await openai.createChatCompletion({
      model: "gpt-3.5-turbo",
      messages: [{ role: "assistant", content: topic }],
    });
    console.log(completion);
    return "world";
  } catch (err) {
    return res.status(500).json({ message: err });
  }
};

// role: "user": a message from the user
//       "assistant": a response from ChatGPT
//       "system": generally one initial message defining How we want ChatGPT to talk

// const gptGetResponse = async (cmd) => {
//   await fetch("https://api.openai.com/v1/chat/completions", {
//     method: "POST",
//     headers: {
//       Authorization: "Bearer " + process.env.OPENAI_API_KEY,
//       "Content-Type": "application/json",
//     },
//     body: JSON.stringify({
//       "model": "gpt-3.5-turbo",
//       "messages": [
//         {
//           role: "system",
//           content:
//             "Critically revise all articles or issues according to the given topic as a professional writing checker.",
//         },
//         cmd,
//       ],
//     }),
//   });
// };

// const gptGetTitle = async (topic) => {
//   await fetch("https://api.openai.com/v1/chat/completions", {
//     method: "POST",
//     headers: {
//       Authorization: "Bearer " + process.env.OPENAI_API_KEY,
//       "Content-Type": "application/json",
//     },
//     body: JSON.stringify({
//       "model": "gpt-3.5-turbo",
//       "messages": [
//         {
//           role: "system",
//           content: "Make the most suitable summary for the given topic.",
//         },
//         topic,
//       ],
//     }),
//   }).then ((data) => {
//     console.log(data.json())
//     return data.choices[0].message.content;
//   }).catch ((err) => {console.log(err)})
// };

module.exports = { gptGetResponse, gptGetTitle };

File: server/controllers/gpt.py
import sys
import openai

openai.api_key = "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"

cmd = sys.argv[1]
temp = sys.argv[2]


while True:
    request = str(input().split("\n"))

    completion = openai.ChatCompletion.create(
        # The main input is the messages parameter
        # Messages must be an array of message objects,
        # where each object has a role (either “system”, “user”, or “assistant”) and content (message content).
        # Conversations can be as short as 1 message or fill many pages.
        # Typically, a conversation is formatted with a system message first, followed by alternating user and assistant messages.
        model="gpt-3.5-turbo",
        messages=[
            {"role": "user", "content": request},
        ],
        temperature=temp,
    )

    print(completion.choices[0].message.content + "\n")
